src/main.rs

```diff
@@ -60,10 +63,49 @@ pub struct UploadResponse {
     url: String,
 }
 
+struct DirEntryModTimePair {
+    dir_entry: DirEntry,
+    mod_time: SystemTime,
+}
+
 async fn index() -> impl Responder {
     HttpResponse::Ok().body("i API ready!")
 }
 
+async fn recent(opt: web::Data<Opt>) -> impl Responder {
+    let mut files = Vec::new();
+
+    let base_dir = match get_base_dir(&opt) {
```

Comment:
This could then be just `let base_dir = get_base_dir(&opt)?;`, much shorter :)

Reply:
Fixed!